blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java

```diff
@@ -838,10 +839,10 @@ public String completeMultipartUpload(MultipartUpload mpu, List<MultipartPart> p
             throw propagate(ioe);
          }
          streams.add(is);
-         partHashes.append(blobPart.getMetadata().getETag());
+         md5Hasher.putBytes(new BigInteger(blobPart.getMetadata().getETag(), 16).toByteArray());
```

Comment:
This is clever but would `BaseEncoding.base16.decode` address this more naturally?